### ISSUE TITLE
gg_raster: drops legend because col/fill added instead of just fill

### ISSUE THREAD

**Issue by davidhodge931** [davidhodge931/ggblanket#334]
``` r
library(tidyverse)

#legend does not work with geom_raster
ggplot2::faithfuld |>
  head(10) |>
  ggplot() +
  geom_raster(aes(x = waiting, y = eruptions, col = density, fill = density))
#> Warning in geom_raster(aes(x = waiting, y = eruptions, col = density, fill =
#> density)): Ignoring unknown aesthetics: colour
```

![](https://i.imgur.com/Z3pBgBk.png)

<sup>Created on 2023-06-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by davidhodge931** [comment]
``` r
library(tidyverse)

#legend works if aesthetic order is fill then col
ggplot2::faithfuld |>
  head(10) |>
  ggplot() +
  geom_raster(aes(x = waiting, y = eruptions, fill = density, col = density))
#> Warning in geom_raster(aes(x = waiting, y = eruptions, fill = density, col =
#> density)): Ignoring unknown aesthetics: colour
```

![](https://i.imgur.com/mhR6B3h.png)

<sup>Created on 2023-06-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by davidhodge931** [comment]
Consider move back to how it was before, if this gets fixed
https://github.com/tidyverse/ggplot2/issues/5324

**Comment by davidhodge931** [comment]
Fixed for gglplot2 3.4.3, so fill can be unhashed in gg_raster without consequence

**Comment by davidhodge931** [comment]
Sorted
